mayastor/src/lvs/lvol.rs

```diff
@@ -330,4 +353,59 @@ impl Lvol {
             }
         }
     }
+
+    /// Format snapshot name
+    /// base_name is the nexus or replica UUID
+    pub fn format_snapshot_name(base_name: &str, snapshot_time: u64) -> String {
+        format!("{}-snap-{}", base_name, snapshot_time)
+    }
+
+    /// Create a snapshot
+    pub async fn create_snapshot(
+        self,
+        nvmf_req: *mut spdk_nvmf_request,
```

Comment:
I'd like to avoid the usage of raw pointers as much as possible. Perhaps we can do something like 

```rust
#[derive(Clone)]
struct NvmfReq(NonNull<*mut spdk_nvmf_request>)
```

Then we can also do nicer things in the future like Impl Debug?

Reply:
Fixed, we agreed it should be `NonNull<spdk_nvmf_request>`.